e2e/loadbalancer_test.go

```diff
@@ -239,38 +220,49 @@ func TestLoadbalancerReservedIP(t *testing.T) {
 
 }
 
-func cleanUp(mirrorDeploy *appsv1.Deployment, svc *corev1.Service) error {
-	e2eTest.tenantClient.Delete(context.TODO(), svc)
-
-	return e2eTest.tenantClient.Delete(context.TODO(), mirrorDeploy)
+func cleanUp(ctx context.Context, mirrorDeploy *appsv1.Deployment, svc *corev1.Service) (err error) {
+	if svc != nil {
+		err = e2eTest.tenantClient.Delete(ctx, svc)
+	}
+	if mirrorDeploy != nil {
+		err = errors.Join(err, e2eTest.tenantClient.Delete(ctx, mirrorDeploy))
+	}
+	return err
 }
 
-func getOrCreateIP(c *civogo.Client) (*civogo.IP, error) {
-	ip, err := c.FindIP("e2e-test-ip")
+func getOrCreateIP(c *civogo.Client) (_ *civogo.IP, cleanup func() error, _ error) {
+	ip, err := c.FindIP("ccm-e2e-test-ip")
 	if err != nil && civogo.ZeroMatchesError.Is(err) {
 		ip, err = c.NewIP(&civogo.CreateIPRequest{
-			Name: "e2e-test-ip",
+			Name:   "ccm-e2e-test-ip",
+			Region: e2eTest.civo.Region,
 		})
 		if err != nil {
-			return nil, err
+			return nil, nil, err
+		}
+		cleanup = func() error {
+			_, err := c.DeleteIP(ip.ID)
+			return err
 		}
 	} else if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
-	return ip, err
+	return ip, cleanup, err
 }
 
-func getOrCreateSvc(c client.Client) (*corev1.Service, error) {
-	svc := &corev1.Service{}
-	err := c.Get(context.TODO(), client.ObjectKey{Name: "echo-pods", Namespace: "default"}, svc)
-	if err != nil && errors.IsNotFound(err) {
+func getOrCreateSvc(ctx context.Context, c client.Client) (svc *corev1.Service, created bool, _ error) {
+	svc = &corev1.Service{}
+	err := c.Get(ctx, client.ObjectKey{Name: "echo-pods", Namespace: "default"}, svc)
+	if err != nil && apierrors.IsNotFound(err) {
 		lbls := map[string]string{"app": "mirror-pod"}
 		// Create a service of type: LoadBalancer
 		svc = &corev1.Service{
 			ObjectMeta: metav1.ObjectMeta{
-				Name:        "echo-pods",
-				Namespace:   "default",
-				Annotations: map[string]string{},
+				Name:      "echo-pods",
+				Namespace: "default",
+				Annotations: map[string]string{
+					"kubernetes.civo.com/firewall-id": e2eTest.cluster.FirewallID,
```

Comment:
>[!NOTE]
Same as the below comment.
https://github.com/civo/civo-cloud-controller-manager/pull/37/files#r2031178002